cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -55,6 +88,13 @@ TEST(GetTakeIndices, Basics) {
     auto indices_array = MakeArray(indices);
     ValidateOutput(indices);
     AssertArraysEqual(*expected_indices, *indices_array, /*verbose=*/true);
+
+    ASSERT_OK_AND_ASSIGN(auto ree_filter, REEncode(*filter));
+    ASSERT_OK_AND_ASSIGN(auto indices_from_ree,
+                         internal::GetTakeIndices(*ree_filter->data(), null_selection));
+    auto indices_from_ree_array = MakeArray(indices);
+    ValidateOutput(indices_from_ree);
+    AssertArraysEqual(*expected_indices, *indices_from_ree_array, /*verbose=*/true);
```

Comment:
This is repeated in subsequent tests. Perhaps factor it out in a utility function or test fixture method?

Reply:
Fixing now.